process/smartContract/hooks/blockChainHook.go

```diff
@@ -440,6 +487,24 @@ func (bh *BlockChainHookImpl) ProcessBuiltInFunction(input *vmcommon.ContractCal
 	return vmOutput, nil
 }
 
+func (bh *BlockChainHookImpl) checkMaxCounters(input *vmcommon.ContractCallInput) error {
```

Comment:
also here, I suggest renaming the function, reversing the if order (this helps reduce the indentation of the inner code) and to protect with a mutex the counters area

Reply:
done